test/plugin/test_out_detect_exceptions.rb

```diff
@@ -98,6 +114,48 @@ def test_exception_detection
     assert_equal(make_logs(t, *messages), d.events)
   end
 
+  def test_ignore_exception_nested_in_text
+    test_cases = {
+      "php": PHP_EXC,
+      "python": PYTHON_EXC,
+      "ruby": RUBY_EXC
+    }
+
+    test_cases.each do |language, exception|
+      cfg = "languages #{language}"
+      d = create_driver(cfg)
+      t = Time.now.to_i
+
+      # Converting exception to a single line to simply the test case
+      single_line_exception = exception.gsub("\n", "\\n")
+
+      # There is a nested exception within the body, we should ignore those!
+      lines = [
+        # Json payload
+        %({"timestamp": {"nanos": 998152494, "seconds": 1496420064}, "message": "#{single_line_exception}", "thread": 139658267147048, "severity": "ERROR"}\n),
+        %({"timestamp": {"nanos": 5990266, "seconds": 1496420065}, "message": "next line", "thread": 139658267147048, "severity": "INFO"}\n),
```

Comment:
I think JSON is a sufficiently important special case to be tested separately. I'd also suggest testing at least prefixed *and* suffixed, though JSON should cover some of that.